src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -209,6 +209,7 @@ public async Task RefreshAsync(CancellationToken cancellationToken)
                     //
                     // Avoid instance state modification
                     Dictionary<string, ConfigurationSetting> applicationSettings = null;
+                    Dictionary<KeyValueIdentifier, ConfigurationSetting> watchedSettings = new Dictionary<KeyValueIdentifier, ConfigurationSetting>(_watchedSettings);

```

Comment:
Why is this initialized here? I'd expect behavior similar to `applicationSettings`